### DISCUSSION TITLE
Template zettels

### DISCUSSION THREAD

**Discussion by srid** [srid/neuron#649] [Ideas]
I imagine we don't need a *very flexible* template system. 

**Initial proposal**, to be refined:

- Templates can be added to ~~a subdirectory, eg: `./templates/daily.md`~~ (see [Revision 2 below](https://github.com/srid/neuron/issues/328#issuecomment-687834345))
- Template vars are simple strings wrapped in `__` (see below)
- Have `neuron new --template=daily` use the specified template

Here's a sample template, `./template/daily.md`:

```markdown
---
date: __DATE__
tags:
  - journal
---

# __TITLE__

## Agenda today

- 

## Achieved today

Today I achieved, ...
```

We may even go one step ahead and support arbitrary key value pairs (`neuron new --template=daily --data="key1=123"`), with `__KEY1__` somewhere in the template body.

**Comment by srid** [comment]
note to self: @TheMatten and I discussed some ideas on supporting *general* template functions (using the `{{...}}` syntax of https://github.com/srid/neuron/issues/312#issuecomment-666313401), and that needs to be spec'ed out one day.

**Comment by bangedorrunt** [comment]
i think we should add `title` in front matter. that would help much when doing `grep` search, current _title_ using `#` might conflict with code where `#` is used for comment.

**Comment by srid** [comment]
@babygau `title` in front matter is already supported. `# title` is used only when you don't specify one in the front matter.

Neuron generates `# title` by default, however - because it is the natural choice when writing Markdown. The grep problem you mention should be resolved on its own (lexical grep?), rather than *requiring* front matter title (which should remain optional) simply because of grep's shortcomings.

Anyway what we are talking about is unrelated to this issue; it should be a discussion in a new issue.

**Comment by bangedorrunt** [comment]
tks u for response. i asked because i came from here ([meta data](https://neuron.zettel.page/2011505.html)) and it doesn't mention _title_. i  also had an warning error complain about malformed data when i have _title:_ metadata

**Comment by srid** [comment]
Guess we should document it then. As for the metadata error, try putting the title in double quotes (having to do this is yet another reason to prefer regular Markdown titles).

**Comment by srid** [comment]
**Revision 2**

This proposal could further be simplied. No need for a hardcoded list files; just put the templates in `neuron.dhall`, which also allows us to support *structured* templates (and then do dhall2yaml for YAML frontmatter). Consider something like [dhall-text](https://hackage.haskell.org/package/dhall-text) if appropriate.
